### DISCUSSION TITLE
📢 PaddleSOT 技术交流会来啦～ 🎉

### DISCUSSION THREAD

**Discussion by SigureMo** [2742195759/paddle-symbolic-trace#149] [General]
## 一、Motivation 🚀

为了回应大家参与 PaddleSOT 的热情，帮助大家更容易上手 PaddleSOT，降低学习成本，📢 我们计划在 **本周四（06.15）** 进行一次技术交流会，欢迎大家在本 Discussion 中提出在阅读代码、开发过程中遇到的问题，我们将会在讨论会上统一为大家进行解答！🤝

**问题收集时间截止到本周三（06.14）晚～**

另外如果大家有想要分享的内容（比如源码阅读心得、自己对现有机制的理解等），也欢迎在下面留言报名 🙋🏻‍♀️～

## 二、Summary 📚

|  关键词 | 问题描述 | 贡献提问者 | 贡献解答者 |
|:-----:|:-----:|:-----:|:-----:|
| 机制 | SOT里触发 break graph 的入口在哪个模块？ | @Aurelius84  |  @0x45f  | 
|  设计    |   "Symbolic OpCode Translator"里的Symbolic， 跟[torch.fx](https://pytorch.org/docs/stable/fx.html) 里提到的Symbolic tracer里提到的Symbolic的区别是什么。    |     @jzhang533   | @2742195759 
|  设计 |   想知道SIR的设计理念和思路，以及为什么没有设计成类似fx graph的IR？     |  @jzhang533     |   @2742195759      |
| 机制 | 可以通过某个测试单例，讲解一下整个执行流程，以及涉及到的一些机制吗。比如是如何做缓存，如何进行字节码的转写，怎样进行模拟执行和模拟调用 |  @gouzil  |  @SigureMo    |
| 设计 | 为什么不在c端处理字节码，类似于写一个自己的Cpython |  @gouzil  |  @2742195759   |
| 设计 | https://github.com/PaddlePaddle/community/tree/master/pfcc/paddle-code-reading/symbolic_opcode_translator 提到 `拓宽接入编译器的路径` ，所以这个项目未来是否会作为 Paddle 对标 Dynamo 的一个基础模块提供给用户接口支持更多后端，而不只是支持动转静 |  @zrr1999 |   @Aurelius84   |
| 机制 | https://github.com/PaddlePaddle/PaddleSOT/pull/141 中遇到了两个问题，一个是触发子图打断时结果跟直观上差距比较大，所以想知道具体是怎么出现的 | @zrr1999  | @0x45f  |
| 机制 | 另一个问题是当编译期没有报错，运行期出现的错误，例如 x[100] 这种越界问题，只有实际运行才会出现，结果会不符合预期，所以想知道这个运行时的异常机制是否有些额外的设计，从而可以更好的处理这种情况，或者未来有没有计划把这种检查加入编译期，之前 @SigureMo 提到后面要添加编译期的检查，但是不知道是否包含这种情况，希望可以顺便介绍一下之前提到的 PyTorch 那边的方案和思路 | @zrr1999 | @SigureMo  |

**Comment by Aurelius84** [comment]
🤔 提问：SOT里触发 break graph 的入口在哪个模块？

**Comment by jzhang533** [comment]
- 🤔 提问："**Symbolic** OpCode Translator"里的Symbolic， 跟[torch.fx](https://pytorch.org/docs/stable/fx.html) 里提到的`Symbolic tracer`里提到的Symbolic的区别是什么。
- 🤔 提问：想知道SIR的设计理念和思路，以及为什么没有设计成类似fx graph的IR？

**Comment by gouzil** [comment]
- 🤔 提问:  可以通过某个测试单例，讲解一下整个执行流程，以及涉及到的一些机制吗。比如是如何做缓存，如何进行字节码的转写，怎样进行模拟执行和模拟调用
- 🤔 提问:  为什么不在c端处理字节码，类似于写一个自己的Cpython

**Comment by zrr1999** [comment]
- 🤔 提问:  https://github.com/PaddlePaddle/community/tree/master/pfcc/paddle-code-reading/symbolic_opcode_translator 提到 `拓宽接入编译器的路径` ，所以这个项目未来是否会作为 Paddle 对标 Dynamo 的一个基础模块提供给用户接口支持更多后端，而不只是支持动转静
- 🤔 提问:  https://github.com/PaddlePaddle/PaddleSOT/pull/141 中遇到了两个问题，一个是触发子图打断时结果跟直观上差距比较大，所以想知道具体是怎么出现的
- 🤔 提问:  另一个问题是当编译期没有报错，运行期出现的错误，例如 x[100] 这种越界问题，只有实际运行才会出现，结果会不符合预期，所以想知道这个运行时的异常机制是否有些额外的设计，从而可以更好的处理这种情况，或者未来有没有计划把这种检查加入编译期，之前 @SigureMo 提到后面要添加编译期的检查，但是不知道是否包含这种情况，希望可以顺便介绍一下之前提到的 PyTorch 那边的方案和思路
